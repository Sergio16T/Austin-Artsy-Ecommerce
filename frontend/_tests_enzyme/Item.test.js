import ItemComponent, { ItemButtons } from '../components/Item'; 
import { shallow } from 'enzyme'; 
import { shallowToJson } from 'enzyme-to-json'; 

//these tests are somewhat brittle. Any changes made to layout etc in item, if I decide to change them will break the test

const mockItem = {
    id: "12eadsfasf",
    title: 'Green Shoe', 
    price: 5000,
    description: "Trainer", 
    image: "shoe.jpg", 
    largeImage: 'largeImage'
}

describe('<Item/>', () => {
    it('renders and matches the snapshot', () => {
        //to update snapshot press u 
        const wrapper = shallow(<ItemComponent item={mockItem}/>)
        expect(shallowToJson(wrapper)).toMatchSnapshot(); 
    }); 
    // it('renders price and title and displays properly', () => {
    //     const wrapper = shallow(<ItemComponent item={mockItem}/>); 
    //     const price = wrapper.find('p').at(1); 
    //     const title  = wrapper.find('a').at(0).text(); 
    //     // console.log(wrapper.debug()); 
    //     //if you need to go one level deeper in another component for example use .dive() or maybe .children
    //     // console.log('price', Price.text()); 
    //     // console.log(title); 
    //     expect(price.text()).toBe("$50.00");
    //     expect(title).toBe(mockItem.title);

    // }); 
    // it('renders images properly', () => { 
    //     const wrapper = shallow(<ItemComponent item={mockItem}/>); 
    //     const img = wrapper.find('img'); 
    //     expect(img.props().src).toBe(mockItem.image);
    //     expect(img.props().alt).toBe(mockItem.title);
    // }); 
    // it('renders button list properly', () => {
    //     const wrapper = shallow(<ItemComponent item={mockItem}/>); 
    //     console.log(wrapper.debug()); 
    //     const buttonList = wrapper.find(ItemButtons); 
    //     console.log(buttonList.debug()); 
    //     expect(buttonList.children()).toHaveLength(3); 
    //     expect(buttonList.find('Link')).toHaveLength(1); 
    //     expect(buttonList.find('AddToCart').exists()).toBe(true); 
    //     expect(buttonList.find('DeleteItem')).toHaveLength(1); 
    // }); 
});