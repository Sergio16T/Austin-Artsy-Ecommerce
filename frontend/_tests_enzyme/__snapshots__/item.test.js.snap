// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__StyledItem>
  <img
    alt="Green Shoe"
    src="shoe.jpg"
  />
  <Item__Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "12eadsfasf",
          },
        }
      }
    >
      <a>
        Green Shoe
      </a>
    </Link>
  </Item__Title>
  <p>
    Trainer
  </p>
  <p>
    $50.00
  </p>
  <Item__ItemButtons>
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "12eadsfasf",
          },
        }
      }
    >
      <a
        id="editButton"
      >
         Edit 
      </a>
    </Link>
    <AddToCart
      id="12eadsfasf"
    />
    <DeleteItem
      id="12eadsfasf"
    >
      Delete
    </DeleteItem>
  </Item__ItemButtons>
</ItemStyles__StyledItem>
`;
